flask_praetorian/base.py

```diff
@@ -48,9 +63,10 @@ def __init__(self, app=None, user_class=None, is_blacklisted=None):
         self.pwd_ctx = None
         self.hash_scheme = None
         self.salt = None
+        self.app = app
 
         if app is not None and user_class is not None:
-            self.init_app(app, user_class, is_blacklisted)
+            self.app = self.init_app(app, user_class, is_blacklisted)
```

Comment:
Binding of app to self is not necessary. Should just be:
`self.init_app(app, user_class, is_blacklisted)`

If the app instance is needed in the Praetorian code, use `flask.current_app` instead

Reply:
reverted

`    def __init__(self, app=None, user_class=None, is_blacklisted=None):
        self.pwd_ctx = None
        self.hash_scheme = None
        self.salt = None

        if app is not None and user_class is not None:
            self.app = self.init_app(app, user_class, is_blacklisted)`